### PR TITLE
compliant and non-compliant code examples documentation

### PR DESCRIPTION
**Pull request by schalkms**
I just started with it. (work in progess)

### CODE REVIEW

**Comment by schalkms** [comment]
@vanniktech I think we already had a conversation about this topic somewhere here. In my opinion it's not worth the effort since you need to resolve and annotate the test cases which should be included in the documentation. You also need to build a generator and so on and so forth.
I also think that this information is not likely to change.

File: detekt-generator/documentation/potential-bugs.md [hunk 1]
```diff
@@ -27,6 +27,25 @@ Flags duplicate case statements in when expressions.
 If a when expression contains the same case statement multiple times they should be merged. Otherwise it might be
 easy to miss one of the cases when reading the code, leading to unwanted side effects.
 
+#### Noncompliant Code:
+
+```kotlin
+when (i) {
```
Comment (Mauin) [inline detekt-generator/documentation/potential-bugs.md]: The first lines in the code blocks should be indented as well, shouldn't it?

Comment (schalkms) [reply detekt-generator/documentation/potential-bugs.md]: Good catch! I will update the generator.

**Comment by schalkms** [comment]
I think this can be merged now.

**Pull request merged**